app/src/main/java/io/appium/uiautomator2/core/EventRegister.java

```diff
@@ -0,0 +1,40 @@
+package io.appium.uiautomator2.core;
+
+
+import android.app.UiAutomation;
+import android.support.test.uiautomator.Configurator;
+import android.view.accessibility.AccessibilityEvent;
+
+import io.appium.uiautomator2.model.AppiumUiAutomatorDriver;
+
+public abstract class EventRegister {
+
+    public static Boolean runAndRegisterScrollEvents (ReturningRunnable<Boolean> runnable) {
+        AccessibilityEvent event = null;
+        UiAutomation.AccessibilityEventFilter eventFilter = new UiAutomation.AccessibilityEventFilter() {
+            @Override
+            public boolean accept(AccessibilityEvent event) {
+                return event.getEventType() == AccessibilityEvent.TYPE_VIEW_SCROLLED;
+            }
+        };
+
+        try {
+            //wait for AccessibilityEvent filter
+            event = UiAutomatorBridge.getInstance().getUiAutomation().executeAndWaitForEvent(runnable,
+                    eventFilter, Configurator.getInstance().getScrollAcknowledgmentTimeout());
+        } catch (Exception ignore) {}
```

Comment:
`executeAndWaitForEvent` throws only one checked exception `TimeoutException`, why we need catch here all the things? Also logging might be useful here.

Reply:
And I'd advise to erase information about the previous event, if we couldn't get the expected one.